daprdocs/content/en/reference/components-reference/supported-name-resolution/nr-sqlite.md

```diff
@@ -0,0 +1,54 @@
+
+{{% alert title="Note" color="primary" %}}
+
+This component is optimized to be used in scenarios where all Dapr instances are running on the same physical machine, where the database is accessed through the same, locally-mounted disk.  
+Using the SQLite nameresolver with a database file accessed over the network (including via SMB/NFS) can lead to issues such as data corruption, and is **not supported**.
+{{% /alert %}}
+
+## Configuration format
+
+Name resolution is configured via the [Dapr Configuration]({{< ref configuration-overview.md >}}).
+
+Within the Configuration YAML, set the `spec.nameResolution.component` property to `"sqlite"`, then pass configuration options in the `spec.nameResolution.configuration` dictionary.
+
+This is the basic example of a Configuration resource:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: appconfig
+spec:
+  nameResolution:
+    component: "sqlite"
+    version: "v1"
+    configuration:
+      connectionString: "/home/user/.dapr/nr.db"
+```
+
+## Spec configuration fields
+
+When using the SQLite component as name resolver, the `spec.nameResolution.configuration` dictionary contains these options:
+
+| Field        | Required | Type | Details  | Examples |
+|--------------|:--------:|-----:|:---------|----------|
+| `connectionString` | Y | `string` | The connection string for the SQLite database. Normally, this is the path to a file on disk, relative to the current working directory, or absolute. | `"nr.db"` (relative to the working directory), `"/home/user/.dapr/nr.db"` |
+| `updateInterval` | N | [Go duration](https://pkg.go.dev/time#ParseDuration) (as a `string`) | Interval for active Dapr sidecars to update their status in the database.<br>Must be at least 1s greater than `timeout`. Values with fractions of seconds are truncated (for example, `1500ms` becomes `1s`). Default: `5s` | `"2s"` |
+| `timeout` | N | [Go duration](https://pkg.go.dev/time#ParseDuration) (as a `string`).<br>Must be at least 1s. | Timeout for operations on the database. Integers are interpreted as number of seconds. Defaults to `1s` | `"2s"`, `2` |
+| `tableName` | N | `string` | Name of the table where the data is stored. Defaults to `hosts`. | `"hosts"` |
+| `metadataTableName` | N | `string` | Name of the table used by Dapr to store metadata for the component. Defaults to `metadata`. | `"metadata"` |
+| `cleanupInterval` | N | [Go duration](https://pkg.go.dev/time#ParseDuration) (as a `string`) | Interval to remove stale records from the database. Default: `1h` (1 hour) | `"10m"` |
+| `busyTimeout` | N | [Go duration](https://pkg.go.dev/time#ParseDuration) (as a `string`) | Interval to wait in case the SQLite database is currently busy serving another request, before returning a "database busy" error. Default: `800ms` (800 milliseconds) | `"100ms"` |
```

Comment:
How does this relate to the timeout above? They sound like they do the same thing. When would I set this over using timeout above. Would be great to have some guidance here in the doc

Reply:
There's some overlap, but they behave differently.

- busy timeout controls how locking works in SQLite. With SQLite, writes are exclusive, so every time any app is writing something, the DB is locked. If another app tries to write, it will wait up to "busy timeout" before returning the "database busy" error
- the timeout option controls the timeout for the entire operation. For example if the query "hangs", after the DB has acquired the lock (so after busy timeout is cleared), then timeout comes into effect

For SQLite, most users should not touch this